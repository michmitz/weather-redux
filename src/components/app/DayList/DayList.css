.listContainer {
  margin: 5px;
}

ul {
  list-style: none;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  text-align: center;
}