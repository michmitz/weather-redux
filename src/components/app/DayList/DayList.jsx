/* eslint-disable max-len */
import React from 'react';
import DayItem from '../DayItem/DayItem';
import { useSelector } from 'react-redux';

const DayList = () => {
  const forecast = useSelector(state => state.weather);
  const loading = useSelector(state => state.loading);

  if(loading) {
    return <h1>Loading...</h1>;
  }

  const dateElements = forecast.map(date => (
    <li key={date.date}>
      <DayItem {...date} />
    </li>
  ));

  return (
    <ul>
      {dateElements}
    </ul>
  );
};

export default DayList;
