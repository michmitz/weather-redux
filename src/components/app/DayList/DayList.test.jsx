import React from 'react';
import { render, cleanup } from '@testing-library/react';
import DayList from './DayList';
import { Provider } from 'react-redux';
import store from '../../../store';

describe('User component', () => {
  afterEach(() => cleanup());
  it('renders DayList', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <DayList />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
