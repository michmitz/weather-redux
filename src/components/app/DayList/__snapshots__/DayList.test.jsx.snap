// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayList component renders DayList 1`] = `
<DocumentFragment>
  <h1>
    Loading...
  </h1>
</DocumentFragment>
`;
